src/scranpy/nearest_neighbors/build_neighbor_index.py

```diff
@@ -12,39 +12,27 @@
 
 
 class NeighborIndex:
-    """Class for nearest neighbor search index.
-
-    Args:
-        ptr (ct.c_void_p): Pointer reference to scran's nearest neighbor index.
+    """The nearest neighbor search index.
+    This should not be manually constructed but should be created
+    by :py:meth:`~scranpy.nearest_neighbors.build_neighbor_index.build_neighbor_index`.
     """
 
     def __init__(self, ptr: ct.c_void_p):
-        """Initialize the class."""
         self.__ptr = ptr
 
     def __del__(self):
-        """Free the reference."""
         lib.free_neighbor_index(self.__ptr)
 
-    @property
-    def ptr(self) -> ct.c_void_p:
-        """Get pointer to scran's NN search index.
-
-        Returns:
-            ct.c_void_p: Pointer reference.
-        """
-        return self.__ptr
```

Comment:
any particular reason why this is deleted?

Reply:
also applied to `NeighborResults` in `find_nearest_neighbors`

Reply:
The raw pointer shouldn't be exposed to users.

In theory, third-party developers of Python packages with C++ extensions could use the pointer to do something, but that's pretty speculative. Probably best to wait until someone wants it.

Reply:
I put these lines back because we also use them in other steps, e.g. https://github.com/BiocPy/scranpy/blob/redoc/src/scranpy/dimensionality_reduction/run_tsne.py#L194

Reply:
the reason to set it up this way was that the `ptr` is only a read-only prop

Reply:
The read-only nature isn't the issue here, it's the fact that it's documented (which implies that it's usable by people other than us). So, an undocumented property would work just as well, but in that case, why not just have our functions directly access it as `__ptr`? (Or don't even name it `__ptr`, and access it as `ptr` directly?)